jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextRequest.java

```diff
@@ -58,7 +65,7 @@ public class ServletContextRequest extends ContextRequest
     public static ServletContextRequest getServletContextRequest(ServletRequest request)
     {
         if (request instanceof ServletApiRequest)
-            return ((ServletApiRequest)request).getServletContextRequest();
+            return ((ServletApiRequest)request).getServletRequestInfo().getServletContextRequest();
```

Comment:
This seems odd. The ServletRequestInfo interface is designed so we don't have to reference the ServletContextRequest directly, and is implemented by the ServletContextRequest. So seems odd that the interface has this method. Is it really necessary?

Reply:
We do need a way to navigate back from a `ServletApiRequest` to the `ServletContextRequest`,  But I have now made that only be `ServletContextRequest.getServletContextRequest(ServletRequest)`